### ISSUE TITLE
Buffer deprecated warning

### ISSUE THREAD

**Issue by sandy081** [thejoshwolfe/yauzl#114]
I am developer from VS Code and we are using this module `yauzl v2.9.2` and it throws following warning.

`(node:336) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.`

This warning is coming from its dependency `fd-slicer`

```
internal/process/warning.js:127
      throw warning;
      ^

DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
    at showFlaggedDeprecation (buffer.js:156:11)
    at new Buffer (buffer.js:171:3)
    at /Applications/Visual Studio Code.app/Contents/Resources/app/node_modules.asar/fd-slicer/index.js:109:18
    at pendGo (/Applications/Visual Studio Code.app/Contents/Resources/app/node_modules.asar/pend/index.js:54:3)
    at onCb (/Applications/Visual Studio Code.app/Contents/Resources/app/node_modules.asar/pend/index.js:41:7)
    at /Applications/Visual Studio Code.app/Contents/Resources/app/node_modules.asar/fd-slicer/index.js:31:7
    at FSReqCallback.wrapper [as oncomplete] (fs.js:477:5) {
  name: 'DeprecationWarning',
  code: 'DEP0005'
}
```

Filed issue against fd-slicer module - https://github.com/andrewrk/node-fd-slicer/issues/3

Ref: https://github.com/microsoft/vscode/issues/82524

**Comment by nalhabash** [comment]
@sandy081 @thejoshwolfe  this issue is still happening for me when i deploy my project using yauzl.
It's because of the fd-slicer dependency using a deprecated constructor.
My PR there fixes the issue: https://github.com/thejoshwolfe/yauzl/pull/115

**Comment by hwlmatt** [comment]
@nalhabash thank you for this [fix](https://github.com/thejoshwolfe/yauzl/pull/115)!

For anyone else, I implemented this temporarily (hoping @nalhabash's will be merged in) by forking `yauzl` and aliasing the broken `fd-slicer` to nalhabash's [`fd-slicer2`](https://github.com/nalhabash/fd-slicer2) package instead. 
Allowed me to proceed without having to make any changes to the actual `yauzl` codebase.

in `package.json` this reads as:
`"dependencies": {
    "fd-slicer": "npm:fd-slicer2",`

To make the change in your own fork:
remove the old one:
`npm uninstall fd-slicer` 
alias with new one: 
`npm install fd-slicer@npm:fd-slicer2`

Thanks again to @nalhabash

**Comment by lensbart** [comment]
@thejoshwolfe Could you please take a look at https://github.com/thejoshwolfe/yauzl/pull/115? This would help with node 16 compatibility (currently the process exits with an error code). Thanks!!!

**Comment by thejoshwolfe** [comment]
Thanks yall for your patience. I have finally found the enthusiasm to address this issue. Fixed in yauzl 3.0.0
